crates/autopilot/src/domain/fee/mod.rs

```diff
@@ -110,9 +129,18 @@ impl ProtocolFees {
             buy: quote.buy_amount,
             fee: quote.fee,
         };
-        let protocol_fees = protocol_fees
+        let apply_protocol_fee = if let Ok(Some(address)) = order
```

Comment:
This could be done better, it is mostly a placeholder to apply different kind of discounts in the future.